addons/web/static/src/js/core/utils.js

```diff
@@ -977,6 +977,24 @@ var utils = {
             ['name', '=like', '%.assets\_%.js'],
         ];
     },
+
+    /*
+     *  Split the node into multiple elements according to its marked down text
+     *  Supported markdowns are
+     *   - ** to set to bold
+     *   - __ to set to italic
+     */
+    split_node_markdown: function(node) {
+        node.css('white-space', 'pre').css('font-weight', 'normal');
+        let html = node.html();
+        // format according to markdown
+        html = html.replace(/(\*\*)(.*?)\1/g, '<b>$2</b>');
+        html = html.replace(/(__)(.*?)\1/g, '<i>$2</i>');
+        // unescape the string
+        html = html.replace(/\\\*/g, '*');
+        html = html.replace(/\\\_/g, '_');
+        node.html(html);
```

Comment:
I think we should avoid using `.html(...)` when possible. While your code looks safe, it's also fragile. A refactoring (i.e. I wouldn't be surprised if we support additional markup in the future) or bad bugfix could be an easy XSS vector. Instead, a safer way would be to extract the parts between marks and manually create the HTML element (`createElement`) and set its content with the `.text(...)` function. This way, even if we miss some escaping somewhere, we are certain it won't be evaluated by the browser.
By the way, you should use `strong` and `em` instead of `b` and `i` (semantic vs rendering).

Reply:
> Instead, a safer way would be to extract the parts between marks and manually create the HTML element (`createElement`) and set its content with the `.text(...)` function. This way, even if we miss some escapin

What? That looks worse in my opinion. I also have a test checking that the xml/html is not parsed.



> By the way, you should use `strong` and `em` instead of `b` and `i` (semantic vs rendering).

I am doing rendering, not semantics.

Reply:
This PR should be closed anyway, as it has been decided that the "feature" requesting this can be dropped.